CMakeLists.txt

```diff
@@ -139,7 +141,7 @@ list(APPEND KNOWHERE_LINKER_LIBS Folly::folly)
 add_library(knowhere SHARED ${KNOWHERE_SRCS})
 add_dependencies(knowhere ${KNOWHERE_LINKER_LIBS})
 if(WITH_RAFT)
-  list(APPEND KNOWHERE_LINKER_LIBS raft::raft CUDA::cublas CUDA::cusparse CUDA::cusolver)
+  list(APPEND KNOWHERE_LINKER_LIBS raft::raft raft::compiled_static CUDA::cublas CUDA::cusparse CUDA::cusolver)
```

Comment:
This helps eliminate unintended compilation of RAFT algorithms for types that RAFT does not optimize for. By building the static libraft ourselves, we can be confident that we are only referencing types that RAFT algorithms handle optimally.